components/tidb_query_expr/src/impl_json.rs

```diff
@@ -159,10 +159,47 @@ pub fn json_merge(args: &[Option<JsonRef>]) -> Result<Option<Json>> {
 #[rpn_fn(writer)]
 #[inline]
 fn json_quote(input: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
-    let str = String::from_utf8(input.to_vec()).map_err(tidb_query_datatype::codec::Error::from)?;
-    let result =
-        Bytes::from(serde_json::to_string(&str).map_err(tidb_query_datatype::codec::Error::from)?);
-    Ok(writer.write(Some(result)))
+    Ok(writer.write(quote(input)?))
+}
+
+fn quote(bytes: BytesRef) -> Result<Option<Bytes>> {
+    let mut result = Vec::with_capacity(bytes.len() * 2 + 2);
+    result.push(b'\"');
+    for byte in bytes.iter() {
+        if *byte == b'\"' || *byte == b'\\' {
+            result.push(b'\\');
+            result.push(*byte)
+        } else if *byte == b'\x07' {
+            // \a alert
+            result.push(b'\\');
+            result.push(b'a');
+        } else if *byte == b'\x08' {
```

Comment:
Maybe we need to export https://github.com/tikv/tikv/blob/master/components/tidb_query_datatype/src/codec/mysql/json/json_unquote.rs#L12 ?

Reply:
`'\x0A'` is same with `b'\n'`.

Reply:
I'm lacking some understanding on `json_quote`, ping @andylokandy to take a look.